agent/command/git.go

```diff
@@ -512,7 +527,9 @@ func (c *gitFetchProject) Execute(ctx context.Context, comm client.Communicator,
 		return errors.Wrap(err, "applying expansions")
 	}
 
-	projectMethod, projectToken, err := getProjectMethodAndToken(c.Token, conf.Expansions.Get(evergreen.GlobalGitHubTokenExpansion), conf.Expansions.Get(evergreen.GithubAppToken), conf.GetCloneMethod())
+	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
+
+	projectMethod, projectToken, err := getProjectMethodAndToken(ctx, comm, td, conf.ProjectRef.Owner, conf.ProjectRef.Repo, c.Token, conf.Expansions.Get(evergreen.GlobalGitHubTokenExpansion), conf.GetCloneMethod())
```

Comment:
this is a _lot_ of variables; can you just make this a private method of c and pass in all of conf to avoid many strings in a row?

Reply:
couldnt make it a method on c because both git fetch and push uses it but passes in conf now